tests/beacon/test_helpers.py

```diff
@@ -729,3 +734,46 @@ def test_get_domain(pre_fork_version,
         slot=current_slot,
         domain_type=domain_type,
     )
+
+
+def test_is_double_vote(sample_attestation_data_params):
+    attestation_data_1_params = {
+        **sample_attestation_data_params,
+        'slot': 12345,
+    }
+    attestation_data_1 = AttestationData(**attestation_data_1_params)
+
+    attestation_data_2_params = {
+        **sample_attestation_data_params,
+        'slot': 12345,
+    }
+    attestation_data_2 = AttestationData(**attestation_data_2_params)
+
+    assert is_double_vote(attestation_data_1, attestation_data_2)
+
+    attestation_data_3_params = {
+        **sample_attestation_data_params,
+        'slot': 54321,
+    }
+    attestation_data_3 = AttestationData(**attestation_data_3_params)
+
+    assert not is_double_vote(attestation_data_1, attestation_data_3)
+
+
+def test_is_surround_vote(sample_attestation_data_params):
+    attestation_data_1_params = {
+        **sample_attestation_data_params,
+        'slot': 4,
+        'justified_slot': 0,
+
+    }
+    attestation_data_1 = AttestationData(**attestation_data_1_params)
+
+    attestation_data_2_params = {
+        **sample_attestation_data_params,
+        'slot': 3,
+        'justified_slot': 2,
+    }
+    attestation_data_2 = AttestationData(**attestation_data_2_params)
+
+    assert is_surround_vote(attestation_data_1, attestation_data_2)
```

Comment:
We can use `@pytest.mark.parametrize` for more test cases. For example:
```python

@pytest.mark.parametrize(
    (
        'attestation_1_slot, attestation_1_justified_slot, attestation_2_slot, attestation_2_justified_slot, expected'
    ),
    [
        (0, 0, 0, 0, False),
        (4, 3, 3, 2, False),  # not (attestation_data_1.justified_slot < attestation_data_2.justified_slot
        (4, 0, 3, 1, False),  # not (attestation_data_2.justified_slot + 1 == attestation_data_2.slot)
        (4, 0, 4, 3, False),  # not (attestation_data_2.slot < attestation_data_1.slot)
        (4, 0, 3, 2, True),

    ],
)
def test_is_surround_vote(sample_attestation_data_params,
                          attestation_1_slot,
                          attestation_1_justified_slot,
                          attestation_2_slot,
                          attestation_2_justified_slot,
                          expected):
    attestation_data_1_params = {
        **sample_attestation_data_params,
        'slot': attestation_1_slot,
        'justified_slot': attestation_1_justified_slot,
     }
    attestation_data_1 = AttestationData(**attestation_data_1_params)
    attestation_data_2_params = {
        **sample_attestation_data_params,
        'slot': attestation_2_slot,
        'justified_slot': attestation_2_justified_slot,
    }
    attestation_data_2 = AttestationData(**attestation_data_2_params)
    assert is_surround_vote(attestation_data_1, attestation_data_2) == expected
```

Reply:
Thanks. I just pushed a commit incorporating your suggestions.